src/position.cpp

```diff
@@ -1296,8 +1298,8 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
   {
       if (is_ok(from) && (gates(us) & from))
           st->gatesBB[us] ^= from;
-      if (type_of(m) == CASTLING)
-          st->gatesBB[us] ^= to;
+      if (type_of(m) == CASTLING && (gates(us) & rfrom))
```

Comment:
Thanks, looks good to me. Just as a minor note, in order to reduce the frequency of merge conflicts from upstream I usually try to avoid touching upstream code if feasible. Here one could use `to_sq(m)` instead of `rfrom` in order to retain the scope of `rfrom`, since SF always uses the rook square as `to_sq` in move encoding (irrespective of whether it is 960 or not), although `rfrom` might be more explicit.

Reply:
Thx! Done.